### ISSUE TITLE
CA 1508 false positive – object assignation inside of task not detected

### ISSUE THREAD

**Issue by bravequickcleverfibreyarn** [dotnet/roslyn-analyzers#5276]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.302](https://dotnet.microsoft.com/download/dotnet/5.0)

_AND_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

Analyzer fails to detect object assignation inside task.

### Sample

```c#
static async Task Method()
{
 object? obj = null;
 
 await Method(async () => await Task.Run(() => obj = new()));
 
 if (obj is null) // Warning CA1508 'obj is null' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
 return;
 
 static Task Method(Func<Task> func) => func();
}

static void Method2()
{
 object? obj = new();
 
 Method(() => Task.Run(() => obj = null).GetAwaiter().GetResult());
 
 if (obj is null) // Warning CA1508 'obj is null' is always 'false'. Remove or refactor the condition(s) to avoid dead code.
     return;
 
 static void Method(Action action) => action();
}
```

### Expected behavior
No CA1508.
### Actual behavior
CA1508 false positive.
## Additional context

- Sample project attached – [6050FalsePositive.zip](https://github.com/dotnet/roslyn-analyzers/files/6861129/6050FalsePositive.zip)
- Upon change in `.csproj` false positive _CA1508_ disappears – try to remove `<OutputType>Exe</OutputType>` in sample project.

**Comment by mavasani** [comment]
Verified this repros on latest .NET 8 preview SDK, but has been fixed with #6811 and does not repro after it.
